### ISSUE TITLE
Pipeline panics requesting webrtcbin pad upon incoming connection

### ISSUE THREAD

**Issue by septatrix** [centricular/webrtcsink#49]
When trying to connect to the stream in a browser I the pipeline panics while trying to request a pad...

https://github.com/centricular/webrtcsink/blob/b51cdb9de0a85ecd90ea0d91742d0b79637ebdfd/plugins/src/webrtcsink/imp.rs#L997-L1000

<details>
<summary>Backtrace</summary>

```
thread 'async-std/runtime' panicked at 'called `Option::unwrap()` on a `None` value', plugins/src/webrtcsink/imp.rs:1000:14
stack backtrace:
   0:     0x7f74cb3c48bc - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h572300f0ebfde2a7
   1:     0x7f74cb3e53ac - core::fmt::write::h9a40954895def713
   2:     0x7f74cb3c0f51 - std::io::Write::write_fmt::h589135a83419f2d6
   3:     0x7f74cb3c61e5 - std::panicking::default_hook::{{closure}}::h619386df4f38a529
   4:     0x7f74cb3c5ea4 - std::panicking::default_hook::ha87f40e82c315143
   5:     0x7f74cb3c6918 - std::panicking::rust_panic_with_hook::h220e25558f92196d
   6:     0x7f74cb3c65e9 - std::panicking::begin_panic_handler::{{closure}}::hffd3696cd0a1a22a
   7:     0x7f74cb3c4d54 - std::sys_common::backtrace::__rust_end_short_backtrace::hbec50169ea271f2f
   8:     0x7f74cb3c6349 - rust_begin_unwind
   9:     0x7f74caf40943 - core::panicking::panic_fmt::h5f7f26e7daf91179
  10:     0x7f74caf4080d - core::panicking::panic::hde4091d9c455f69f
  11:     0x7f74cb06e8bc - core::option::Option<T>::unwrap::h2b7c83dfa6a81e50
                               at /builddir/build/BUILD/rustc-1.60.0-src/library/core/src/option.rs:752:21
  12:     0x7f74cb057c87 - webrtcsink::webrtcsink::imp::Consumer::request_webrtcbin_pad::ha9bc3330545d381a
                               at /home/septatrix/Documents/university/hiwi/webrtcsink/plugins/src/webrtcsink/imp.rs:997:19
  13:     0x7f74caff75a9 - webrtcsink::webrtcsink::imp::WebRTCSink::add_consumer::{{closure}}::hcfb5aaf65f72bda6
                               at /home/septatrix/Documents/university/hiwi/webrtcsink/plugins/src/webrtcsink/imp.rs:1672:37
  14:     0x7f74cb0e045c - core::iter::traits::iterator::Iterator::for_each::call::{{closure}}::hb7a9a0d906cc6c04
                               at /builddir/build/BUILD/rustc-1.60.0-src/library/core/src/iter/traits/iterator.rs:770:29
  15:     0x7f74cb1492c1 - core::iter::traits::iterator::Iterator::fold::hf31f1cf70cf8316c
                               at /builddir/build/BUILD/rustc-1.60.0-src/library/core/src/iter/traits/iterator.rs:2285:21
  16:     0x7f74cb1495d3 - core::iter::traits::iterator::Iterator::for_each::hc8cd90657463fafd
                               at /builddir/build/BUILD/rustc-1.60.0-src/library/core/src/iter/traits/iterator.rs:773:9
  17:     0x7f74cb05df3a - webrtcsink::webrtcsink::imp::WebRTCSink::add_consumer::h1a74dee9f4bd9acd
                               at /home/septatrix/Documents/university/hiwi/webrtcsink/plugins/src/webrtcsink/imp.rs:1669:9
  18:     0x7f74cb0fdf53 - webrtcsink::webrtcsink::WebRTCSink::add_consumer::hc4f31bfe0ca77c26
                               at /home/septatrix/Documents/university/hiwi/webrtcsink/plugins/src/webrtcsink/mod.rs:121:9
  19:     0x7f74cb089afa - webrtcsink::signaller::imp::Signaller::connect::{{closure}}::{{closure}}::h4ac6514f6c7a0c6f
                               at /home/septatrix/Documents/university/hiwi/webrtcsink/plugins/src/signaller/imp.rs:130:59
  20:     0x7f74cb0297ec - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h9e43e22d93401936
                               at /builddir/build/BUILD/rustc-1.60.0-src/library/core/src/future/mod.rs:84:19
  21:     0x7f74cafb0a4b - <async_std::task::builder::SupportTaskLocals<F> as core::future::future::Future>::poll::{{closure}}::h5139028be468c2b8
                               at /home/septatrix/.cargo/registry/src/github.com-1ecc6299db9ec823/async-std-1.10.0/src/task/builder.rs:199:17
  22:     0x7f74cafaf477 - async_std::task::task_locals_wrapper::TaskLocalsWrapper::set_current::{{closure}}::h59911093f14587f9
                               at /home/septatrix/.cargo/registry/src/github.com-1ecc6299db9ec823/async-std-1.10.0/src/task/task_locals_wrapper.rs:60:13
  23:     0x7f74caf9f76c - std::thread::local::LocalKey<T>::try_with::h977d6f66f357ff64
                               at /builddir/build/BUILD/rustc-1.60.0-src/library/std/src/thread/local.rs:413:16
  24:     0x7f74caf9d83e - std::thread::local::LocalKey<T>::with::hceb3cd4fa7f2065b
                               at /builddir/build/BUILD/rustc-1.60.0-src/library/std/src/thread/local.rs:389:9
  25:     0x7f74cafaedb9 - async_std::task::task_locals_wrapper::TaskLocalsWrapper::set_current::h673b3bf171e63e98
                               at /home/septatrix/.cargo/registry/src/github.com-1ecc6299db9ec823/async-std-1.10.0/src/task/task_locals_wrapper.rs:55:9
  26:     0x7f74cafb0667 - <async_std::task::builder::SupportTaskLocals<F> as core::future::future::Future>::poll::ha4d5e2bdb44799a8
                               at /home/septatrix/.cargo/registry/src/github.com-1ecc6299db9ec823/async-std-1.10.0/src/task/builder.rs:197:13
  27:     0x7f74cafdf6f1 - async_executor::Executor::spawn::{{closure}}::hbba5c600ce9d51e3
                               at /home/septatrix/.cargo/registry/src/github.com-1ecc6299db9ec823/async-executor-1.4.1/src/lib.rs:144:19
  28:     0x7f74cb02a5cd - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::hf1046bc8e5055a6f
                               at /builddir/build/BUILD/rustc-1.60.0-src/library/core/src/future/mod.rs:84:19
  29:     0x7f74cb0b963b - async_task::raw::RawTask<F,T,S>::run::h72b6ad9a001b7e8d
                               at /home/septatrix/.cargo/registry/src/github.com-1ecc6299db9ec823/async-task-4.2.0/src/raw.rs:489:20
  30:     0x7f74cb2d8321 - async_task::runnable::Runnable::run::h35584324ff2d9bad
                               at /home/septatrix/.cargo/registry/src/github.com-1ecc6299db9ec823/async-task-4.2.0/src/runnable.rs:309:18
  31:     0x7f74cb28da91 - async_executor::Executor::run::{{closure}}::{{closure}}::h2ef4ffbd52b4656e
                               at /home/septatrix/.cargo/registry/src/github.com-1ecc6299db9ec823/async-executor-1.4.1/src/lib.rs:235:21
  32:     0x7f74cb28110d - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h0cca0c85da8e38e9
                               at /builddir/build/BUILD/rustc-1.60.0-src/library/core/src/future/mod.rs:84:19
  33:     0x7f74cb28621b - <futures_lite::future::Or<F1,F2> as core::future::future::Future>::poll::hd1bc901109009cff
                               at /home/septatrix/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-lite-1.12.0/src/future.rs:529:33
  34:     0x7f74cb28cedd - async_executor::Executor::run::{{closure}}::h478e49eff0617cc0
                               at /home/septatrix/.cargo/registry/src/github.com-1ecc6299db9ec823/async-executor-1.4.1/src/lib.rs:242:31
  35:     0x7f74cb2812bd - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h3bb5d16482c6ec09
                               at /builddir/build/BUILD/rustc-1.60.0-src/library/core/src/future/mod.rs:84:19
  36:     0x7f74cb2862bb - <futures_lite::future::Or<F1,F2> as core::future::future::Future>::poll::hd94ad6e1d044780b
                               at /home/septatrix/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-lite-1.12.0/src/future.rs:529:33
  37:     0x7f74cb29cce2 - async_io::driver::block_on::h7484ac9586ff33e7
                               at /home/septatrix/.cargo/registry/src/github.com-1ecc6299db9ec823/async-io-1.6.0/src/driver.rs:142:33
  38:     0x7f74cb2a31c0 - async_global_executor::reactor::block_on::{{closure}}::hb9aad0c4a96ac7e5
                               at /home/septatrix/.cargo/registry/src/github.com-1ecc6299db9ec823/async-global-executor-2.0.3/src/reactor.rs:3:18
  39:     0x7f74cb2a301d - async_global_executor::reactor::block_on::h420355991122187f
                               at /home/septatrix/.cargo/registry/src/github.com-1ecc6299db9ec823/async-global-executor-2.0.3/src/reactor.rs:12:5
  40:     0x7f74cb282c34 - async_global_executor::threading::thread_main_loop::{{closure}}::{{closure}}::hd9bcebd97855b19e
                               at /home/septatrix/.cargo/registry/src/github.com-1ecc6299db9ec823/async-global-executor-2.0.3/src/threading.rs:95:17
  41:     0x7f74cb2a8f71 - std::thread::local::LocalKey<T>::try_with::h964cbe591043fbfe
                               at /builddir/build/BUILD/rustc-1.60.0-src/library/std/src/thread/local.rs:413:16
  42:     0x7f74cb2a84b4 - std::thread::local::LocalKey<T>::with::hdbc85aa70334b10e
                               at /builddir/build/BUILD/rustc-1.60.0-src/library/std/src/thread/local.rs:389:9
  43:     0x7f74cb282af0 - async_global_executor::threading::thread_main_loop::{{closure}}::hc5e95cffff0c157b
                               at /home/septatrix/.cargo/registry/src/github.com-1ecc6299db9ec823/async-global-executor-2.0.3/src/threading.rs:89:13
  44:     0x7f74cb285acd - std::panicking::try::do_call::h17d066c8e5811a8c
                               at /builddir/build/BUILD/rustc-1.60.0-src/library/std/src/panicking.rs:492:40
  45:     0x7f74cb28653b - __rust_try
  46:     0x7f74cb2859a4 - std::panicking::try::he234b6cae888a135
                               at /builddir/build/BUILD/rustc-1.60.0-src/library/std/src/panicking.rs:456:19
  47:     0x7f74cb2a0cbf - std::panic::catch_unwind::had83c38a8085df66
                               at /builddir/build/BUILD/rustc-1.60.0-src/library/std/src/panic.rs:137:14
  48:     0x7f74cb2828ee - async_global_executor::threading::thread_main_loop::hc8e4a9c0d76ba7ce
                               at /home/septatrix/.cargo/registry/src/github.com-1ecc6299db9ec823/async-global-executor-2.0.3/src/threading.rs:88:12
  49:     0x7f74cb28f9ba - core::ops::function::FnOnce::call_once::h88428eb583cf2b7b
                               at /builddir/build/BUILD/rustc-1.60.0-src/library/core/src/ops/function.rs:227:5
  50:     0x7f74cb2a0c4a - std::sys_common::backtrace::__rust_begin_short_backtrace::hb7bc374d4ad07430
                               at /builddir/build/BUILD/rustc-1.60.0-src/library/std/src/sys_common/backtrace.rs:122:18
  51:     0x7f74cb2a6747 - std::thread::Builder::spawn_unchecked_::{{closure}}::{{closure}}::h4bcb774117d8cccb
                               at /builddir/build/BUILD/rustc-1.60.0-src/library/std/src/thread/mod.rs:498:17
  52:     0x7f74cb2a5a3a - <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once::hbfb01a8517f1cdfd
                               at /builddir/build/BUILD/rustc-1.60.0-src/library/core/src/panic/unwind_safe.rs:271:9
  53:     0x7f74cb285afe - std::panicking::try::do_call::ha73cfb7f8457ed1c
                               at /builddir/build/BUILD/rustc-1.60.0-src/library/std/src/panicking.rs:492:40
  54:     0x7f74cb28653b - __rust_try
  55:     0x7f74cb2858f1 - std::panicking::try::hcedc5cb65a342d50
                               at /builddir/build/BUILD/rustc-1.60.0-src/library/std/src/panicking.rs:456:19
  56:     0x7f74cb2a0c8a - std::panic::catch_unwind::h6fa584fdf6cecdfb
                               at /builddir/build/BUILD/rustc-1.60.0-src/library/std/src/panic.rs:137:14
  57:     0x7f74cb2a6512 - std::thread::Builder::spawn_unchecked_::{{closure}}::hec25a22de0814f0c
                               at /builddir/build/BUILD/rustc-1.60.0-src/library/std/src/thread/mod.rs:497:30
  58:     0x7f74cb28f88f - core::ops::function::FnOnce::call_once{{vtable.shim}}::h3286e7717c216e7d
                               at /builddir/build/BUILD/rustc-1.60.0-src/library/core/src/ops/function.rs:227:5
  59:     0x7f74cb3c93e3 - std::sys::unix::thread::Thread::new::thread_start::h079e1972701670ee
  60:     0x7f74d9853b1a - start_thread
  61:     0x7f74d98d8660 - __GI___clone3
  62:                0x0 - <unknown>
```

</details>

**Comment by sdroege** [comment]
Can you attach a debug log produced with `GST_DEBUG=webrtc*:9`? Thanks

**Comment by septatrix** [comment]
> Can you attach a debug log produced with `GST_DEBUG=webrtc*:9`? Thanks

That debug hint was all that I needed. I saw some complaints about a missing libnice component and after installing `libnice-gstreamer1.0` (Fedora) it worked as expected. Thanks :)

Also I got everything working with the gstreamer from the fedora repos so I am not sure how relevant the section about having to compile gstreamer-main is. If it is possible to easily detect missing stuff like the libnice component maybe add some feature detection before start and if not all features are available abort and only then tell the user to use a newer version/compile from source?
